app/models/payment_intent.rb

```diff
@@ -1,12 +1,15 @@
 # frozen_string_literal: true
 
 class PaymentIntent < ApplicationRecord
-  SUCCEEDED_STATES = %w[succeeded].freeze
+  SUCCEEDED_STATES = [Finance::StripeChargeService::PAYMENT_INTENT_SUCCEEDED].freeze
 
   belongs_to :invoice, inverse_of: :payment_intents
 
-  validates :invoice, :payment_intent_id, :state, presence: true
-  validates :payment_intent_id, :state, length: { maximum: 255 }
+  self.ignored_columns = %w[payment_intent_id]
+
+  validates :invoice, :reference, :state, presence: true
+  validates :reference, :state, length: { maximum: 255 }
+  validates :reference, uniqueness: true
```

Comment:
```suggestion
  validates :reference, uniqueness: true, presence: true
```

Reply:
And remove 2 lines above?

Reply:
Honestly I don't see the need to change this. Now it's like the validations are grouped by rule, rather than by attribute.

Reply:
aaah sorry sorry, I had not seen that it was validated above 🙈

Reply:
in my defense, my left arm is being hunted at the moment 🐈